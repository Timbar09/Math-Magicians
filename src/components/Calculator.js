import '../calculator.css';
import { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const btnNames = [
    'AC',
    '+/-',
    '%',
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '7',
    '8',
    '9',
    '0',
    '.',
    'รท',
    'ร',
    '-',
    '+',
    '=',
  ];
  const numbers = btnNames.slice(0, 14);
  const operators = btnNames.slice(14);

  const [state, setState] = useState({
    total: null,
    next: null,
    operator: null,
  });

  const handleClick = (btnName) => {
    setState(calculate(state, btnName));
  };

  const updateOutput = () => {
    const { total, next, operator } = state;

    if (operator) return `${total ?? '0'} ${operator} ${next ?? ''}`;

    if (next) return next;

    return total || 0;
  };

  return (
    <main className="calculator">
      <header className="calc-output">{updateOutput()}</header>

      <div className="calc-grid">
        <section className="calc-numbers">
          {numbers.map((number) => (
            <button
              onClick={() => handleClick(number)}
              key={number}
              className={`btn btn-${number}`}
              type="button"
            >
              {number}
            </button>
          ))}
        </section>

        <section className="calc-operators">
          {operators.map((operator) => (
            <button
              onClick={() => handleClick(operator)}
              key={operator}
              className={`btn btn-${operator}`}
              type="button"
            >
              {operator}
            </button>
          ))}
        </section>
      </div>
    </main>
  );
};

export default Calculator;
