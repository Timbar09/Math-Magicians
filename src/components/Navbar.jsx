import styles from '../styles/Navbar.module.css';

const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];

const Navbar = () => (
  <header className={styles.header}>
    <nav className={`${styles.navbar} container padding-x`}>
      <span>
        <a href={links[0].path} className={styles.logo}>
          Math MAgicians
        </a>
      </span>
      <ul className={styles['nav-links']}>
        {links.map((link) => (
          <li key={link.text}>
            <a href={link.path} className={styles['nav-link']}>
              {link.text}
            </a>
          </li>
        ))}
      </ul>
    </nav>
  </header>
);

export default Navbar;
