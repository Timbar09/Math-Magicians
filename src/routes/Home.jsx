const Home = () => (
  <main>
    <div className="container padding">
      <h1>Welcome to our page</h1>

      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi vitae delectus tempore
        architecto eum accusamus eaque optio nemo unde quia officia iste, aspernatur suscipit
        eligendi. Minima minus magni excepturi dolore?
      </p>

      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Reprehenderit voluptatibus dolorum
        soluta adipisci facere quae dolore corporis esse deleniti quo beatae, molestias minus
        blanditiis et voluptatum odit assumenda quibusdam! Corporis, exercitationem vitae explicabo
        aliquam dolorum inventore eius expedita, aliquid tempora eum delectus aspernatur animi
        alias.
      </p>
    </div>
  </main>
);

export default Home;
