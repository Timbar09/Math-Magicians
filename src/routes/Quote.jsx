const mainStyles = {
  maxHeight: '25rem',
  height: '100vh',
  display: 'grid',
  placeItems: 'center',
};

const quoteStyles = {
  fontSize: '1.25rem',
  maxWidth: '80%',
  margin: 'auto',
};

const quotes = [
  'To forgive is to set a prisoner free and discover that the prisoner was you. - Lewis B. Smedes',
  "Happiness is not the absence of problems, it's the ability to deal with them",
  'It is time you understand that true spirituality means that you yourself become the living expression of the divine teaching you follow. - Omraam Mikhaël Aïvanhov',
  'To a disciple who was forever complaining about others, the Master said, ‘If it is peace you want, seek to change yourself, not other people. It is easier to protect your feet with slippers than to carpet the whole of the earth.’ - Anthony de Mello',
  "If the path before you is clear, you're probably on someone else's. - Joseph Campbell",
  'There are people who make no mistakes because they never wish to do anything worth doing. - Goethe',
  'Happiness is like a butterfly: the more you chase it, the more it will evade you, but if you notice the other things around you, it will gently come and sit on your shoulder. - Henry David Thoreau',
  'Time is the king of all men, he is their parent and their grave, and gives them what he will and not what they crave. - Pericles',
  "You might feel dumb asking questions, but you look dumber when you don't get it because you failed to ask.",
];
const quote = quotes[Math.floor(Math.random() * quotes.length + 1)];

const Quote = () => (
  <main style={mainStyles} className="container padding">
    <p style={quoteStyles}>{`"${quote}"`}</p>
  </main>
);

export default Quote;
