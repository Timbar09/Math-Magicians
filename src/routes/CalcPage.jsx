import Calculator from '../components/Calculator';

const CalcPage = () => (
  <div className="wrapper">
    <Calculator />
  </div>
);

export default CalcPage;
