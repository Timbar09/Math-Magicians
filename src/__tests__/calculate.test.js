import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';
import CalcPage from '../routes/CalcPage';
import Navbar from '../components/Navbar';
import Home from '../routes/Home';
import Quote from '../routes/Quote';
import calculate from '../logic/calculate';

describe('Calculator', () => {
  const obj = { total: null, next: null, operation: null };

  it('matches DOM Snapshot', () => {
    const domTree = renderer.create(<Calculator />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  it('renders Calculator component', () => {
    render(<CalcPage />);
    const linkElement = screen.getByText(/do some math/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('check one of the calculator button exists', () => {
    render(<Calculator />);
    const btn = screen.getByText('1');
    expect(btn).toBeInTheDocument();
  });

  it('Builds the snapchot of the Calculator component safely', () => {
    const rend = renderer.create(<Calculator />);
    expect(rend.toJSON()).toMatchSnapshot();
  });

  it('Render Calculator component', () => {
    render(<Calculator />);
  });

  it('Return null object if it is AC', () => {
    const result = calculate(obj, 'AC');
    expect(result).toEqual({ total: null, next: null, operation: null });
  });

  it('Builds the snapchot of the home component safely', () => {
    const rend = renderer.create(<Home />);
    expect(rend.toJSON()).toMatchSnapshot();
  });

  it('Builds the snapchot of the Navbar component safely', () => {
    const rend = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(rend.toJSON()).toMatchSnapshot();
  });
});
