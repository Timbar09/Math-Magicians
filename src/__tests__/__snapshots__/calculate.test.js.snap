// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Builds the snapchot of the Calculator component safely 1`] = `
<div
  className="calculator"
>
  <header
    className="calc-output"
  >
    0
  </header>
  <div
    className="calc-grid"
  >
    <section
      className="calc-numbers"
    >
      <button
        className="btn btn-AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn-+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn-%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn-1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn-2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn-3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn-4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn-5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn-6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn-7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn-8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn-9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn-0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn-."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </section>
    <section
      className="calc-operators"
    >
      <button
        className="btn btn-÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="btn btn-×"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
      <button
        className="btn btn--"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn-+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn-="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </section>
  </div>
</div>
`;

exports[`Calculator Builds the snapchot of the Navbar component safely 1`] = `
<header
  className="header"
>
  <nav
    className="navbar container padding-x"
  >
    <span>
      <a
        aria-current="page"
        className="logo active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderTop": "none",
          }
        }
      >
        Math MAgicians
      </a>
    </span>
    <ul
      className="nav-links"
    >
      <li>
        <a
          aria-current="page"
          className="({
            isActive
          }) => isActive ? 'active' : undefined nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="({
            isActive
          }) => isActive ? 'active' : undefined nav-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="({
            isActive
          }) => isActive ? 'active' : undefined nav-link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Calculator Builds the snapchot of the Quote component safely 1`] = `
<main
  className="container padding"
  style={
    Object {
      "display": "grid",
      "height": "100vh",
      "maxHeight": "25rem",
      "placeItems": "center",
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "1.25rem",
        "margin": "auto",
        "maxWidth": "80%",
      }
    }
  >
    "To a disciple who was forever complaining about others, the Master said, ‘If it is peace you want, seek to change yourself, not other people. It is easier to protect your feet with slippers than to carpet the whole of the earth.’ - Anthony de Mello"
  </p>
</main>
`;

exports[`Calculator Builds the snapchot of the home component safely 1`] = `
<main>
  <div
    className="container padding"
  >
    <h1>
      Welcome to our page
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi vitae delectus tempore architecto eum accusamus eaque optio nemo unde quia officia iste, aspernatur suscipit eligendi. Minima minus magni excepturi dolore?
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Reprehenderit voluptatibus dolorum soluta adipisci facere quae dolore corporis esse deleniti quo beatae, molestias minus blanditiis et voluptatum odit assumenda quibusdam! Corporis, exercitationem vitae explicabo aliquam dolorum inventore eius expedita, aliquid tempora eum delectus aspernatur animi alias.
    </p>
  </div>
</main>
`;

exports[`Calculator Matches DOM Snapshot 1`] = `
<div
  className="calculator"
>
  <header
    className="calc-output"
  >
    0
  </header>
  <div
    className="calc-grid"
  >
    <section
      className="calc-numbers"
    >
      <button
        className="btn btn-AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn-+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn-%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn-1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn-2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn-3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn-4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn-5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn-6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn-7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn-8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn-9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn-0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn-."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </section>
    <section
      className="calc-operators"
    >
      <button
        className="btn btn-÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="btn btn-×"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
      <button
        className="btn btn--"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn-+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn-="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </section>
  </div>
</div>
`;

exports[`Calculator matches DOM Snapshot 1`] = `
<div
  className="calculator"
>
  <header
    className="calc-output"
  >
    0
  </header>
  <div
    className="calc-grid"
  >
    <section
      className="calc-numbers"
    >
      <button
        className="btn btn-AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn-+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn-%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn-1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn-2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn-3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn-4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn-5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn-6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn-7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn-8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn-9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn-0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn-."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </section>
    <section
      className="calc-operators"
    >
      <button
        className="btn btn-÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="btn btn-×"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
      <button
        className="btn btn--"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn-+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn-="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </section>
  </div>
</div>
`;
